internal/controller/runtime/fsm/runtime_fsm_waiting_shoot_creation.go

```diff
@@ -43,15 +42,15 @@ func sFnWaitForShootCreation(_ context.Context, m *fsm, s *systemState) (stateFn
 		return updateStatusAndRequeueAfter(m.RCCfg.GardenerRequeueDuration)
 
 	case gardener.LastOperationStateFailed:
-		if gardenerhelper.HasErrorCode(s.shoot.Status.LastErrors, gardener.ErrorInfraRateLimitsExceeded) {
-			m.log.Info(fmt.Sprintf("Error during cluster provisioning: Rate limits exceeded for Shoot %s, scheduling for retry", s.shoot.Name))
+		lastErrors := s.shoot.Status.LastErrors
+		reason := imgardenerhandler.ToErrReason(lastErrors...)
+
+		if imgardenerhandler.IsRetryable(lastErrors) {
```

Comment:
General comment: It is strange that inside this method the code `ErrorInfraRateLimitsExceeded` we treated as Retryable is also considered as NON Retryable. But we must assume we did not use this code properly. This code was basically copy & pasted from old Provisioer